### DISCUSSION TITLE
PHP 8 has broken WC

### DISCUSSION THREAD

**Discussion by liderbug** [craigk5n/webcalendar#394] [General]
WC 1.9.1
My hosting company upgraded PHP from 7.4 to 8.1 and the calendar has some problems.  Anyone else have the same problem?  Are there plans to modify WC to work with PHP 8.1? 8.2?
IE password - had to set my password to admin, login, change pw.  logout, login "incorrect pw".  And the link upper right "Login"/"Logout" - stuck on Logout.

**Comment by craigk5n** [comment]
Yes, we have not updated WebCalendar to support PHP 8 yet.  I'm trying to get it done as time permits, but I don't have a specific scheduled completion time yet.

**Reply by liderbug** [reply]
One thing I have found:   $x = substr ($avar, 1,2)  now must be substr ("$avar", ...  You have to quote the var.  
Why? Because "That's the way 'I' like to code...".

**Comment by craigk5n** [comment]
The latest release (v1.9.11 right now) is PHP 8.0 or later and it no longer supports PHP 7.
